### ISSUE TITLE
SnapToSeries on JSON based chart

### ISSUE THREAD

**Issue by CarlosHdez** [amcharts/amcharts4#967]
Seems to me like is not possible to configure a cursor to have `snapToSeries` in a JSON based configuration, I have the following
```
this.chart = am4core.createFromConfig({
  ...
 cursor: {
   ...,
   snapToSeries:  // <--- What goes here, I tried the series id or series JSON but neither work
 }
}, 'id', am4charts.XYChart)
```

I was able to make it work by creating the chart and immediatelly after doing `this.chart.cursor.snapToSeries = this.chart.series.values[0]` but seems dirty

**Comment by martynasma** [comment]
Thanks for the heads up.

Looks like an oversight on our part.

Will be fixing it.

**Comment by martynasma** [comment]
Fixed in [4.1.8](https://github.com/amcharts/amcharts4/releases/tag/4.1.8).

## [4.1.8] - 2019-02-15

### Added
- JSON: You can now specify both list of items and template settings for `ListTemplate`, e.g.: `titles: { template: { ... }, values: [{ ... }] }`. Specifying it the old way will still work.

### Changed
- `DateAxis` will now pay attention to `dateFormatter.firstDayOfWeek` when grid is in "weekly" mode.

### Fixed
- Using `proprtyFields` on a `LineSeries` sometimes could result in an error. (fix by @AndiLi99)
- French locale (fr_FR) updated for correct decimal/thousands separators.
- Export: `scale` in image export options was being ignored.
- JSON: `cursor.snapToSeries` was not working in JSON configs.
- Selection was acting funky when zooming `DateAxis` and cursor moved out of plot area.
- An infinite error message loop fixed which was happening if series was added and there was no X or Y axes defined.
- Tooltip on the last available data item was left visible even if cursor moved to the date where the series had no data points.
- When `snapToSeries` was enabled in `XYCursor`, vertical line was not shown.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.

**Comment by CarlosHdez** [comment]
Thanks for the fast response and solution.
